src/libtriton/arch/x86/x86Semantics.cpp

```diff
@@ -5417,11 +5420,13 @@ namespace triton {
             auto& src  = inst.operands[1];
             auto  op1  = this->symbolicEngine->getOperandAst(inst, dst);
             auto  op2  = this->symbolicEngine->getOperandAst(inst, src);
-            auto  node = this->astCtxt->bvmul(this->astCtxt->sx(dst.getBitSize(), op1), this->astCtxt->sx(src.getBitSize(), op2));
-            auto  expr = this->symbolicEngine->createSymbolicExpression(inst, this->astCtxt->extract(dst.getBitSize()-1, 0, node), dst, "IMUL operation");
+            auto  node1 = this->astCtxt->bvmul(op1, op2);
+            auto  node2 = this->astCtxt->bvmul(this->astCtxt->sx(dst.getBitSize(), op1), this->astCtxt->sx(src.getBitSize(), op2));
+            auto  node3 = (dst.getBitSize() == src.getBitSize()) ? node1 : this->astCtxt->extract(dst.getBitSize()-1, 0, node2);
```

Comment:
I think you can remove this condition (`node3` will always be `node1`). For the `IMUL` with two operands, `src` and `dst` are always of the same size and you cannot have a constant in `dst`.

* `0F AF /r | IMUL r16, r/m16`
* `0F AF /r | IMUL r32, r/m32`
* `REX.W + 0F AF /r | IMUL r64, r/m64`